src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -1777,6 +1834,19 @@ def update_vm_properties(self, agentpool: AgentPool) -> AgentPool:
             agentpool.mode = mode
         return agentpool
 
+    def update_network_profile(self, agentpool: AgentPool) -> AgentPool:
+        self._ensure_agentpool(agentpool)
+
+        asg_ids = self.context.get_asg_ids()
+        allowed_host_ports = self.context.get_allowed_host_ports()
+        if asg_ids or allowed_host_ports:
+            agentpool.network_profile = self.models.AgentPoolNetworkProfile()
+        if asg_ids is not None:
+            agentpool.network_profile.application_security_groups = asg_ids
+        if allowed_host_ports is not None:
+            agentpool.network_profile.allowed_host_ports = allowed_host_ports
```

Comment:
Does this profile support partial update? For example, for an existing nodepool network profile, both `application_security_groups` and `allowed_host_ports` have been configured. Now the user only specifies `application_security_groups` when updating. Then the request body will be `application_security_groups=new value, allowed_host_ports=null`. Is this a valid scenario and the behavior expected?

Reply:
> Does this profile support partial update? 

No. If this feature is turned on, you’ve got to set both values when updating the network profile. You can change ASGs from your own to a managed one if you don’t specify ‘–asg-ids’, and also the other way round.